src/FieldType/EncryptedDatetime.php

```diff
@@ -42,6 +42,8 @@ public function setValue($value, $record = null, $markChanged = true)
 
     public function getDecryptedValue($value)
     {
+        // Type hardening for PHP 8.1+
+        $value = (string)$value;
```

Comment:
Wouldn't it be better to require it to be a string from the start in the method calls? And require it to be a string
`public function getdecryptedvalue(string $value = '') {}`

Reply:
@Firesphere the `getDecryptedValue` functions are only ever called from the `getValue` function within the same class.
The only logic within the `getValue` functions are to return the `getDecryptedValue` function call.
Moving the type hardening from `getDecryptedValue` into `getValue` splits the logic across the 2 functions and raises the question of if the `getDecryptedValue` functions are necessary. 
Is there any reason that the logic within `getDecryptedValue` could not be placed directly in `getValue()`?

Reply:
I'm curious why a simple hardening check (value isn't null, and if it is, make it an empty string), moving it to `getValue()` would suddenly "split the logic".

Given the method is _public_, hardening at call time instead of at execution time, makes more sense to me, as well as requiring a hardened call.

Further, at the `getValue()` point, if the value already is null, it is known that there is nothing to decrypt, thus even calling the method is somewhat pointless at that stage.

Unrelated further, the code duplication smells like it could be moved to a Trait to reduce the amount of copy-paste code.